packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -626,6 +626,15 @@ export class DataStores implements IDisposable {
         }
         return outboundRoutes;
     }
+
+    /**
+     * Returns the package path of the data store with the given id, if the data store exists.
+     */
+    public getDataStorePackagePath(id: string): readonly string[] | undefined {
+        const internalId = this.aliasMap.get(id) ?? id;
+        const context = this.contexts.get(internalId);
+        return context?.packagePath;
```

Comment:
Can we ever have undefined here? If not, please add assert.

Reply:
@vladsud Note that the packagePath will not be available if the data store has not been loaded. I updated the code to handle the scenario.
So we may see `inactiveObjectChanged` events without the packagePath if the data store has not loaded. But if the data store keeps sending ops, then it will be summarized in the next summary and then we will have the packagePath next time this happens.

Reply:
This might be not ideal. Can we make it work for all cases? It would need to be async, but we can easily resolve factory if we want to.

Reply:
It's not straightforward to do this on receiving ops for couple of reasons because:
1. processing op is synchronous.
2. We will have to filter it for summarizer clients only since we don't want other clients to load data stores on receiving ops. Even in the summarizer, loading data stores on receiving ops may not be a good idea perf wise.

We can probably queue these events and send them at summary / GC time later. By that time the data store will have loaded since it received an op and we can get its package data. It just makes the system more complicated though.

Reply:
Makes sense. Given that we use internal IDs (root should almost never get here), we can do joins to get path as long as some event somewhere will have it. So maybe that's the angle to attack - ensure that we always have an event like that?

Reply:
Good idea. We can log an event for every node (or just the data store) when it becomes unreferenced. Since we will log the inactive events for unreferenced nodes only, we should always have this data.

Reply:
I decided to add the events to a queue and log them after GC runs if the package data is unavailable. The reason is that a second event that logs the package data and id has to be after the inactive event. If we log it earlier (like I suggested in my comment), it may be deleted from kusto by the time the inactive event is fired, i.e., if 30 days have passed between them.
There are ways to work around that but the queue felt simpler for now.